posix/unistd.c

```diff
@@ -0,0 +1,609 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/corelibc/errno.h>
+#include <openenclave/corelibc/limits.h>
+#include <openenclave/corelibc/pthread.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/corelibc/string.h>
+#include <openenclave/corelibc/sys/stat.h>
+#include <openenclave/corelibc/sys/utsname.h>
+#include <openenclave/corelibc/unistd.h>
+#include <openenclave/internal/posix/device.h>
+#include <openenclave/internal/posix/fdtable.h>
+#include <openenclave/internal/posix/raise.h>
+#include <openenclave/internal/time.h>
+#include <openenclave/internal/trace.h>
+#include "mount.h"
+#include "posix_t.h"
+
+int oe_gethostname(char* name, size_t len)
+{
+    int ret = -1;
+    struct oe_utsname uts;
+
+    if ((ret = oe_uname(&uts)) != 0)
+        OE_RAISE_ERRNO(oe_errno);
+
+    oe_strlcpy(name, uts.nodename, len);
```

Comment:
oe_strlcpy can return an error

Reply:
We want to allow truncation as specified by the **gethostname()** manual page.